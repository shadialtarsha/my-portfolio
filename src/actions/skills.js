// import fetch from 'node-fetch';

export const setSkills = (skills = []) => ({
  type: 'SET_SKILLS',
  skills,
});

export const startSetSkills = () => dispatch => {
  const url = 'https://shadi-my-portfolio.herokuapp.com/api/skills';
  return fetch(url)
    .then(response => response.json())
    .then(data => dispatch(setSkills(data.skills)))
    .catch(e => console.log(e));
};
