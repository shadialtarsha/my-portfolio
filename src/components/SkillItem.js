import React from 'react';

const SkillItem = ({ skill }) => (
  <div>
    {skill.name} - {skill.percentage}
  </div>
);

export default SkillItem;
