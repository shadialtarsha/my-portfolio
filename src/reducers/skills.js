const skillsReducerDefaultState = [];

const skillsReducer = (state = skillsReducerDefaultState, action) => {
  switch (action.type) {
    case 'SET_SKILLS':
      return action.skills;
    default:
      return state;
  }
};

export default skillsReducer;
