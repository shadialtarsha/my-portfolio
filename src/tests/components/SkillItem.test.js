import React from 'react';
import { shallow } from 'enzyme';
import SkillItem from '../../components/SkillItem';
import skills from '../fixtures/skills';

test('should render skillItem correctly', () => {
  const wrapper = shallow(<SkillItem skill={skills[0]} />);
  expect(wrapper).toMatchSnapshot();
});
