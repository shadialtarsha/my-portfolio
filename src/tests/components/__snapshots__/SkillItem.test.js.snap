// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render skillItem correctly 1`] = `
<div>
  HTML
   - 
  75%
</div>
`;
